traits/has_traits.py

```diff
@@ -124,11 +124,22 @@ class AbstractViewElement(abc.ABC):
     "_trait"
 )
 
+# But whitelist these because they have been sanctioned to be okay
+# prior to the prefix check is put in-place
+WhitelistedNames = (
+    "trait_context",
+    "traits_view",
+    "traits_init",
+)
+
 
 def _is_disallowed_prefix(name):
     """ Returns True if name has a prefix which is in DisallowedNamePrefixes
     """
-    return name.startswith(DisallowedNamePrefixes)
+    return (
```

Comment:
It's tempting to rename this function and reverse the logic here: i.e., call it `_valid_trait_name` rather than `_is_disallowed_prefix`. But it's a private function, so we can do that some other time.